forge-ai/src/main/java/forge/ai/AiAttackController.java

```diff
@@ -142,7 +137,7 @@ public boolean apply(Card c) {
             }
         };
         for (Card c : CardLists.filter(defender.getCardsIn(ZoneType.Battlefield), canAnimate)) {
-            if (c.isToken() && c.getCopiedPermanent() == null) {
+            if (c.isToken() && c.getCopiedPermanent() == null && !c.canTransform(null)) {
```

Comment:
we might check if this logic is a bit outdated?
e.g. _Lita, Mechanical Engineer_ can create Vehicle token

Reply:
Yeah, I think maybe it's possible to remove it altogether?.. It was probably done for speed reasons originally though (checking it on hundreds of tokens in a token deck might be costly, I guess :) )

Reply:
Commented out this code for now, will test with it disabled from now on.

Reply:
So far, testing doesn't indicate any issues with this code commented out.